elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexTransaction.java

```diff
@@ -150,40 +141,52 @@ protected DataStoreTransaction getTransaction(Class<?> cls) {
     }
 
     @Override
-    public <T> Object getRelation(
-            Object entity,
-            RelationshipType relationshipType,
-            String relationName,
-            Class<T> relationClass,
-            EntityDictionary dictionary,
-            Set<Predicate> filters
-    ) {
-        DataStoreTransaction transaction = getTransaction(entity.getClass());
-        return transaction.getRelation(entity, relationshipType, relationName, relationClass, dictionary, filters);
+    public Object getRelation(DataStoreTransaction relationTx,
+                              Object entity,
+                              String relationName,
+                              Optional<FilterExpression> filterExpression,
+                              Optional<Sorting> sorting,
+                              Optional<Pagination> pagination,
+                              RequestScope scope) {
+        com.yahoo.elide.core.RequestScope requestScope;
+        try {
+            requestScope  = (com.yahoo.elide.core.RequestScope) scope;
+        } catch (ClassCastException e) {
+            throw new ClassCastException("Fail trying to cast requestscope");
+        }
+        EntityDictionary dictionary = requestScope.getDictionary();
+        Class<?> relationClass = dictionary.getParameterizedType(entity, relationName);
+        DataStoreTransaction subTransaction = getTransaction(relationClass);
+        return relationTx.getRelation(subTransaction, entity,
```

Comment:
This isn't quite right.  You need to get two different transactions here - the relation transaction and the one for the entity.  Then you would call: entityTransaction.getRelation(relationTransaction, ....)